src/Rector/AbstractDrupalCoreRector.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DrupalRector\Rector;
+
+use DrupalRector\Contract\DrupalCoreRectorInterface;
+use PhpParser\Node;
+use Rector\Core\Rector\AbstractRector;
+use Rector\PostRector\Collector\NodesToAddCollector;
+
+abstract class AbstractDrupalCoreRector extends AbstractRector implements DrupalCoreRectorInterface
+{
+
+
+    /**
+     * @readonly
+     * @var \Rector\PostRector\Collector\NodesToAddCollector
+     */
+    private $nodesToAddCollector;
+
+    public function __construct(NodesToAddCollector $nodesToAddCollector)
+    {
+        $this->nodesToAddCollector = $nodesToAddCollector;
+    }
+
+    public function refactor(Node $node)
+    {
+        if (version_compare(\Drupal::VERSION, $this->getVersion(), '<')) {
+            return null;
+        }
+        $result = $this->doRefactor($node);
+        if ($result === null) {
+            return $result;
+        }
+        $versionCompare = $this->nodeFactory->createFuncCall(
+            'version_compare',
+            [
+                $this->nodeFactory->createClassConstFetch(\Drupal::class, 'VERSION'),
+                $this->getVersion(),
+                '>=',
+            ]
+        );
+        $if = new Node\Stmt\If_($versionCompare, [
+            'stmts' => [new Node\Stmt\Expression($result)],
+            'else' => new Node\Stmt\Else_([
+                new Node\Stmt\Expression($node)
+            ]),
+        ]);
+        $this->nodesToAddCollector->addNodeBeforeNode($if, $node);
```

Comment:
This doesn't work as the `$result` and `$node` are just the replaced parts but ignore the fact it's part of an assignment operation or other context. I think we really do need a PostRector.